autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/AppInstancePatcher.java

```diff
@@ -51,22 +55,46 @@ private void patchInstances(final String appId, final int instances, final boole
-            final HttpPatch patch = new HttpPatch(uriBuilder.build());
-            patch.setEntity(new StringEntity(appArray.toString(), ContentType.APPLICATION_JSON));
-            final HttpResponse response = client.execute(patch);
-            if(!force && response.getStatusLine().getStatusCode()==409){
-                patchInstances(appId, instances, true);
-                return;
-            }
-            if(response.getStatusLine().getStatusCode()!=200){
-                throw new ScalerException(response.getStatusLine().getReasonPhrase());
+        patchInstance(appArray, appId, instances, force);
+    }
+    
+    private void patchInstance(final JsonArray appArray, final String appId, final int instances, final boolean force)
+        throws ScalerException
+    {
+        int count = 0;
+        boolean patched = false;
+        while (!patched) {
+            try (final CloseableHttpClient client = HttpClientBuilder.create().build()) {
+                final URIBuilder uriBuilder = new URIBuilder(marathonUri).setPath("/v2/apps");
+                uriBuilder.setParameters(Arrays.asList(new BasicNameValuePair("force", Boolean.toString(force))));
+                final HttpPatch patch = new HttpPatch(uriBuilder.build());
+                patch.setEntity(new StringEntity(appArray.toString(), ContentType.APPLICATION_JSON));
+                final HttpResponse response = client.execute(patch);
+                if (!force && response.getStatusLine().getStatusCode() == 409) {
+                    patchInstances(appId, instances, true);
+                    return;
+                }
+                if (response.getStatusLine().getStatusCode() != 200) {
+                    if (response.getStatusLine().getStatusCode() == 503) {
+                        LOG.error(String.format("A temporary error occured while attempting to patch service %s. "
+                            + "Patch request returned a 503 status, patching will be reattempted.", appId));
+                        count++;
+                        final int sleepTime = count * 1000;
+                        final int timeToSleep = sleepTime < 10000 ? sleepTime : 10000;
+                        Thread.sleep(timeToSleep);
+                    } else {
+                        LOG.error("Response code: " + response.getStatusLine().getStatusCode());
+                        LOG.error("Response Phrase: " + response.getStatusLine().getReasonPhrase());
+                        throw new ScalerException(response.getStatusLine().getReasonPhrase());
+                    }
+                }
+            } catch (final URISyntaxException | IOException ex) {
+                throw new ScalerException(String.format("Exception patching %s to %s instances.", appId, instances), ex);
+            } catch (final InterruptedException ex) {
+                throw new RuntimeException(ex);

```

Comment:
You're catching the `InterruptedException` but not resetting the interrupt.

Reply:
@dermot-hardy I have updated this PR